### PR TITLE
Add mpulse currency conversion (updated PR)

### PR DESCRIPTION
**Pull request by fklasen**
This is the new PR for the mPulse example.

### CODE REVIEW

**Review by mryakan — CHANGES_REQUESTED** [review]

File: edgekv/examples/mpulse-currency-converter-with-edgekv/main.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+async function getCurrency(request){
+	//in this example the currency code from the purchase order is pulled from a query parameter
+    let params = new URLSearchParams(request.query);
+    let currencyCode   = params.get('currency'); 
+    let default_value  = 1;
+    let currencyValue  = 1;
+	
+    try {
+      const edgeKv = new EdgeKV({namespace: "name", group: "name"}); //replace name and name with your own values
```
[the remark this answers is not in the document]
Comment (fklasen) [reply edgekv/examples/mpulse-currency-converter-with-edgekv/main.js:21]: added the actual names I've used in the implementation

[the remark this answers is not in the document]
Comment (fklasen) [reply edgekv/examples/mpulse-currency-converter-with-edgekv/main.js:21]: done

File: edgekv/examples/mpulse-currency-converter-with-edgekv/main.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    
+    return currencyValue;
+}
+
+export async function responseProvider (request) {
+
+	this.request = request;
+	var finalCurrency= await getCurrency(request);
+	finalCurrency = JSON.parse(finalCurrency)
+
```
[the remark this answers is not in the document]
Comment (fklasen) [reply edgekv/examples/mpulse-currency-converter-with-edgekv/main.js:36]: fixed

File: edgekv/examples/mpulse-currency-converter-with-edgekv/main.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+	class HTMLStream {
+		constructor () { 
+		    let readController = null;
+		    //in this example, the script added modifies the HTML (the div) to visualise that the conversion worked, but in a production system, this would be hidden from the user
+		    //the order value is pulled from the HTML
+		    var script = '<div id="normalized">	</div><script>'+
+		    'var ordervalue= document.getElementById("ordervalue").innerHTML;'+
+		    'var currency = ordervalue.slice(-3);'+
+		    'var amount = ordervalue.slice(0, -3);'+
+		    'var normalized ='+finalCurrency.value+'*amount;'+
```
[the remark this answers is not in the document]
Comment (fklasen) [reply edgekv/examples/mpulse-currency-converter-with-edgekv/main.js:46]: done

File: edgekv/examples/mpulse-currency-converter-with-edgekv/main.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+	}
+    
+    return currencyValue;
+}
+
+export async function responseProvider (request) {
+
+	this.request = request;
+	var finalCurrency= await getCurrency(request);
+	finalCurrency = JSON.parse(finalCurrency)
```
[the remark this answers is not in the document]
Comment (fklasen) [reply edgekv/examples/mpulse-currency-converter-with-edgekv/main.js]: changed

File: edgekv/examples/mpulse-currency-converter-with-edgekv/main.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    let currencyCode   = params.get('currency'); 
+    let default_value  = 1;
+    let currencyValue  = 1;
+	
+    try {
+      const edgeKv = new EdgeKV({namespace: "name", group: "name"}); //replace name and name with your own values
+      currencyValue = await edgeKv.getText({item: currencyCode, default_value: default_value});
+        
+    } catch (error) {
+        return error.toString();
```
[the remark this answers is not in the document]
Comment (fklasen) [reply edgekv/examples/mpulse-currency-converter-with-edgekv/main.js]: changed

File: edgekv/examples/mpulse-currency-converter-with-edgekv/readme.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+The EdgeWorker takes the order value and currency symbol, fetches the conversion value from EdgeKV and finally injects a javascript that performs the normaliziation in the end-user browser.
+
+Please include edgekv.js manually from the parent repository. https://github.com/akamai/edgeworkers-examples/blob/master/edgekv/lib/edgekv.js
+
+## Working POC
+- [mPulse currency converter](http://poc.klasen.se/projects/ew/mpulse-currency-normalizer.php)
+ 
+The POC actually utilizes two EdgeWorkers, one with the EdgeKV but also a variant where the conversion value is fetched from an external service with httpRequest. The only difference in the EdgeWorker codes is in the getCurrency function:
+
+```javascript
```
[the remark this answers is not in the document]
Comment (fklasen) [reply edgekv/examples/mpulse-currency-converter-with-edgekv/readme.md:19]: let's remove it for now, I might add the other example later

**Comment by fklasen** [comment]
all requested changes applied

**Pull request closed without merging**